game-core/src/main/java/games/strategy/triplea/ui/history/HistoryLog.java

```diff
@@ -259,18 +259,11 @@ public void printRemainingTurn(final HistoryNode printNode, final boolean verbos
                 final Collection<Unit> allUnitsInDetails = (Collection<Unit>) details;
                 // purchase/place units - don't need details
                 Unit unit = (Unit) obj;
-                if (title.matches("\\w+ buy .*")) {
+                if (title.matches("\\w+ buy .*") || title.matches("\\w+ attack with .*") || title
+                    .matches("\\w+ defend with .*")) {
```

Comment:
Expressions like this would be a bit more readable as

```java
if (title.matches("\\w+ buy .*")
    || title.matches("\\w+ attack with .*")
    || title.matches("\\w+ defend with .*")) {
```

Reply:
Yip, the parens will get added in automatically. There are *lots* of places where we can use those, it's going to blow up the diff. My plan was to submit that in its own PR. For now I'd like to avoid hand modifications unless it si for correctness, as-is; this PR was 100% machine done.